test/anyons.jl

```diff
@@ -0,0 +1,18 @@
+using DFTK
+using LinearAlgebra
+using Test
+
+@testset "Anyons" begin
+    # Test that the magnetic field satisfies ∇∧A = 2π ρref, ∇⋅A = 0
+    x = 1.23
+    y = -1.8
+    ε = 1e-8
+    M = 2.31
+    σ = 1.81
+    dAdx = (DFTK.magnetic_field_produced_by_ρref(x+ε, y, M, σ) - DFTK.magnetic_field_produced_by_ρref(x, y, M, σ))/ε
+    dAdy = (DFTK.magnetic_field_produced_by_ρref(x, y+ε, M, σ) - DFTK.magnetic_field_produced_by_ρref(x, y, M, σ))/ε
```

Comment:
Lines too long.

Reply:
Frankly I'm not going to bother with style like that in tests

Reply:
Frankly I do. It's a classical broken windows thing. Test code is code and should be as readable as the main code. Either we care about formatting or we don't. If we do we also care in tests.

Reply:
well about formatting you know my answer ;-) I agree that we should treat tests seriously, but I'm already having a very hard time caring about long lines in main code, so in tests...

Reply:
also for this one line breaking is pretty bad... I'll do it for the other one

Reply:
Talking about line breaks ... this is one of the cases where I really would insist. It's easy to miss the ε because the line is so long. Nicer is:
```suggestion
    dAdx = (  DFTK.magnetic_field_produced_by_ρref(x + ε, y, M, σ)
            - DFTK.magnetic_field_produced_by_ρref(x,     y, M, σ)) / ε
    dAdy = (  DFTK.magnetic_field_produced_by_ρref(x, y + ε, M, σ)
            - DFTK.magnetic_field_produced_by_ρref(x, y,     M, σ)) / ε
```
which also makes the structure of what you are doing a lot clearer.

Reply:
I don't really agree, but it's test so I'm less willing to fight, committed your suggestion (thanks to github's new API)